pulsar-metadata/src/main/java/org/apache/pulsar/metadata/api/MetadataCache.java

```diff
@@ -102,6 +102,22 @@ public interface MetadataCache<T> {
      */
     CompletableFuture<T> readModifyUpdateOrCreate(String path, Function<Optional<T>, T> modifyFunction);
 
+    /**
+     * Perform an atomic read-modify-update of the value with the cached version.
+     * <p>
+     * This fail-fasts without retries (e.g. fail-fast upon BadVersionException)
+     * <p>
+     * If the object does not exist yet, the <code>modifyFunction</code> will get passed an {@link Optional#empty()}
+     * object.
+     *
+     * @param path
+     *            the path of the object in the metadata store
+     * @param modifyFunction
+     *            a function that will be passed the current value and returns a modified value to be stored
+     * @return a future to track the completion of the operation
+     */
+    CompletableFuture<T> readModifyUpdateOrCreateOnce(String path, Function<Optional<T>, T> modifyFunction);
```

Comment:
It would be better to add a config (e.g. `maxRetries`) to `MetadataCacheConfig` rather than adding a new API.

Currently we already have a `retryBackoff` config to control the retry backoff. But it cannot prevent the 1st retry.

Reply:
With a config, it will be more flexible to users that they can choose to implement their own retry mechanism or leverage the built-in retry mechanism.

Reply:
updated